src/ert/ensemble_evaluator/snapshot.py

```diff
@@ -122,68 +118,61 @@ def update_metadata(self, metadata: Dict[str, Any]) -> None:
         snapshot's metadata"""
         self._metadata.update(_filter_nones(metadata))
 
-    def update_step(
-        self, real_id: str, step_id: str, step: "Step"
-    ) -> "PartialSnapshot":
-        step_idx = (real_id, step_id)
+    def update_step(self, real_id: str, step: "Step") -> "PartialSnapshot":
         step_update = _filter_nones(
             {
                 "status": step.status,
                 "start_time": step.start_time,
                 "end_time": step.end_time,
             }
         )
-        self._step_states[step_idx].update(step_update)
+        self._step_states[real_id].update(step_update)
         if self._snapshot:
-            self._snapshot._my_partial._step_states[step_idx].update(step_update)
-        self._check_state_after_step_update(step_idx[0], step_idx[1])
+            self._snapshot._my_partial._step_states[real_id].update(step_update)
+        self._check_state_after_step_update(real_id)
         return self
 
     def update_job(
         self,
         real_id: str,
-        step_id: str,
         job_id: str,
         job: "Job",
     ) -> "PartialSnapshot":
-        job_idx = (real_id, step_id, job_id)
+        job_idx = (real_id, job_id)
         job_update = _filter_nones(job.dict())
 
         self._job_states[job_idx].update(job_update)
         if self._snapshot:
             self._snapshot._my_partial._job_states[job_idx].update(job_update)
         return self
 
-    def _check_state_after_step_update(
-        self, real_id: str, step_id: str
-    ) -> "PartialSnapshot":
-        step = self._step_states[(real_id, step_id)]
+    def _check_state_after_step_update(self, real_id: str) -> "PartialSnapshot":
```

Comment:
Does this function even make sense?

Reply:
It is propagating state upwards to the Realization. It should definitely go away at some point, but maybe not in this PR.

Reply:
changed!

Reply:
Will remove this function in a future PR.